// react-animated-buttons/dist/components/constants/animations.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.brigthOut = exports.brigthIn = exports.pulseOut = exports.pulseIn = exports.decrescendo = exports.crescendo = void 0;

var _styledComponents = require("styled-components");

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  0%   {    filter: brightness(1.1);}\n  100% {    filter: brightness(1);}\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  0%   {filter: brightness(1);}\n  50% {filter: brightness(1.3);}\n  100% {filter: brightness(1);}\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  0%   {transform: scale(1.1);}\n  100% {transform: scale(1);}\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  0%   {transform: scale(1);}\n  50% {transform: scale(1.1);}\n  100% {transform: scale(1);}\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  0%   {transform: scale(1.2);}\n  100% {transform: scale(1);}\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  0%   {transform: scale(1);}\n  100% {transform: scale(1.2);}\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var crescendo = (0, _styledComponents.keyframes)(_templateObject());
exports.crescendo = crescendo;
var decrescendo = (0, _styledComponents.keyframes)(_templateObject2());
exports.decrescendo = decrescendo;
var pulseIn = (0, _styledComponents.keyframes)(_templateObject3());
exports.pulseIn = pulseIn;
var pulseOut = (0, _styledComponents.keyframes)(_templateObject4());
exports.pulseOut = pulseOut;
var brigthIn = (0, _styledComponents.keyframes)(_templateObject5());
exports.brigthIn = brigthIn;
var brigthOut = (0, _styledComponents.keyframes)(_templateObject6());
exports.brigthOut = brigthOut;
